feat: add reservation get API

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 -->

- Closes #41 

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- 예약 단건 조회 API 구현
- 호스트/게스트 권한 검증 로직 추가
- N+1 문제 해결을 위한 Fetch Join 쿼리 최적화
- Post 연관관계 매핑 추가

## 🛠️ PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성

zerone1202 committed:
feat: add ReservationGetResponse DTO

- 예약 단건 조회 응답 DTO 생성
- Record 타입과 Builder 패턴 적용
- 예약 ID, 게시글 ID, 호스트/게스트 사용자 ID, 예약 기간, 가격, 보증금, 상태 필드 포함

zerone1202 committed:
feat: add ReservationSuccessMessage constants

- 예약 관련 성공 메시지 상수 클래스 생성
- GET_RESERVATION 메시지 정의

zerone1202 committed:
feat: add ReservationErrorCode enum

- ErrorCode 인터페이스 구현
- RESERVATION_NOT_FOUND 에러 코드 추가 (404)
- RESERVATION_FORBIDDEN 에러 코드 추가 (403)

zerone1202 committed:
feat: add ReservationException class

- GlobalException을 상속한 예약 도메인 예외 클래스 생성
- ReservationErrorCode를 받아 예외 처리

zerone1202 committed:
feat: add ReservationQueryService interface

- 예약 조회 서비스 인터페이스 생성
- findById 메서드 정의 (예약 ID와 현재 사용자 ID로 조회)

zerone1202 committed:
feat: implement reservation findById with authorization

- 예약 단건 조회 비즈니스 로직 구현
- 예약 존재 여부 검증 추가
- 호스트/게스트 권한 검증 로직 추가
- ReservationGetResponse DTO 변환 로직 구현

zerone1202 committed:
feat: add ReservationController interface

- 예약 컨트롤러 인터페이스 생성
- getReservation 메서드 정의 (예약 단건 조회)
- AuthUser를 통한 인증 사용자 정보 전달

zerone1202 committed:
feat: implement reservation GET API endpoint

- GET /api/v1/reservations/{reservationId} 엔드포인트 구현
- @AuthenticationPrincipal을 통한 사용자 인증 처리
- ReservationQueryService를 통한 조회 로직 호출
- CommonApiResponse로 성공 응답 반환

zerone1202 committed:
feat: add Post association to Reservation entity

- Post와 ManyToOne 연관관계 매핑 추가
- post_id 외래키 컬럼 설정
- FetchType.LAZY 적용
- Builder에 Post 파라미터 추가

zerone1202 committed:
feat: add findByIdWithDetails query to prevent N+1

- Fetch Join을 사용한 커스텀 쿼리 메서드 추가
- Post, HostUser, GuestUser를 한 번의 쿼리로 조회
- N+1 문제 해결 (4개 쿼리 → 1개 쿼리로 최적화)
- @Query와 JPQL JOIN FETCH 활용

zerone1202 committed:
feat: apply fetch join query for reservation lookup

- findById()를 findByIdWithDetails()로 변경
- 연관된 Post, HostUser, GuestUser 데이터 즉시 로딩
- 지연 로딩으로 인한 추가 쿼리 발생 방지

zerone1202 committed:
Merge branch 'dev' of https://github.com/sparta-finale/golfrental into feat/41-reservation
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

AllSungho commented on src/main/java/com/golfRental/domain/reservation/repository/ReservationRepository.java:
```diff
@@ -2,6 +2,17 @@
+           "JOIN FETCH r.hostUser " +
+           "JOIN FETCH r.guestUser " +
+           "WHERE r.id = :reservationId")
+    Optional<Reservation> findByIdWithDetails(@Param("reservationId") Long reservationId);
```
N+1 문제 방지를 위해서 작성하신 쿼리 잘 보았습니다!
추가적으로 소프트 딜리트 검증을 위해 deleted_at IS NULL 조건문을 추가하는 것이 어떨까 싶습니다!!

zerone1202 replied:
앗 너무 좋은 리뷰네요! 반영하겠습니다~

zerone1202 committed:
refactor: add soft delete validation in reservation query

- findByIdWithDetails 쿼리에 deletedAt IS NULL 조건 추가
- 삭제된 예약은 조회되지 않도록 개선
- 데이터 무결성 보장 및 소프트 딜리트 정책 준수

AllSungho reviewed (approved):
코드 잘 봤습니다!!!
고생하셨습니다!!!!
